v-next/hardhat-utils/src/fs.ts

```diff
@@ -241,19 +241,22 @@ export async function readJsonFileAsStream<T>(
 
     return result;
   } catch (e) {
-    ensureError<NodeJS.ErrnoException>(e);
+    ensureError(e);
```

Comment:
Can't you use `ensureNodeErrnoExceptionError` to avoid the `if ("code" in e)` check?

Reply:
The `e` here could be either thrown by `new Error` or by `fs` , so I don't think we can use `ensureNodeErrnoExceptionError` there instead.

Reply:
And you’re absolutely right :) I didn’t notice the `new Error` in the function code.